databroker/assets/base_registry.py

```diff
@@ -588,6 +588,7 @@ def bulk_ingest_datum(self, resource_uid, datum_doc_iterable,
     # ## OLD API
     def insert_resource(self, spec, resource_path, resource_kwargs, root=None,
                         path_semantics='posix', uid=None, run_start=None,
+                        id=None,
```

Comment:
Won't it collide with Python's built-in [`id`](https://docs.python.org/3/library/functions.html#id)?

Second question: is it used at all in this PR or anywhere else in the code?

Also, formatting suggestion: this line can be merged with the next line to save space and make it look nicer.

Reply:
It should only collide within the function, where is isn't used.
This is used so we can pass resources in without causing issues, since resources have an id field.

Reply:
@CJ-Wright, thanks for the explanation here and in person. Do you also mind to add the docstring here as well so that the both participating parties have the documentation? Thanks.